ouroboros-network/protocol-tests/Ouroboros/Network/Protocol/TxSubmission/Direct.hs

```diff
@@ -39,8 +40,7 @@ directPipelined (TxSubmissionServerPipelined mserver)
         SendMsgReplyTxIds (BlockingReply txids) client' -> do
           server' <- serverNext txids
           directSender q server' client'
-        SendMsgDone b ->
-          return (a, b)
+        SendMsgDone b -> (,b) <$> a
```

Comment:
Is this change really needed?

Reply:
Yes, because `a :: m a`, while it used to be `a :: a`; using monadic action allows to log a message when sending `MsgDone`.